pkg/scheduler/scheduler_test.go

```diff
@@ -2299,14 +2301,97 @@ func TestSchedulerSchedulePod(t *testing.T) {
+			registerPlugins: []st.RegisterPluginFunc{
+				st.RegisterQueueSortPlugin(queuesort.Name, queuesort.New),
+				st.RegisterPreFilterPlugin(
+					"FakePreFilter1",
+					st.NewFakePreFilterPlugin("FakePreFilter1", nil, nil),
+				),
+				st.RegisterPreFilterPlugin(
+					"FakePreFilter2",
+					st.NewFakePreFilterPlugin("FakePreFilter2", &framework.PreFilterResult{NodeNames: sets.NewString("node2")}, nil),
+				),
+				st.RegisterPreFilterPlugin(
+					"FakePreFilter3",
+					st.NewFakePreFilterPlugin("FakePreFilter3", &framework.PreFilterResult{NodeNames: sets.NewString("node1")}, nil),
+				),
+				st.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
+			},
+			nodes: []string{"node1", "node2", "node3"},
+			pod:   &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "test-prefilter", UID: types.UID("test-prefilter")}},
+			wErr: &framework.FitError{
+				Pod:         &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "test-prefilter", UID: types.UID("test-prefilter")}},
+				NumAllNodes: 3,
+				Diagnosis: framework.Diagnosis{
+					NodeToStatusMap: framework.NodeToStatusMap{
+						"node1": framework.NewStatus(framework.Unschedulable, "node(s) didn't fit the combination of plugin(s) [FakePreFilter2 FakePreFilter3]"),
+						"node2": framework.NewStatus(framework.Unschedulable, "node(s) didn't fit the combination of plugin(s) [FakePreFilter2 FakePreFilter3]"),
+						"node3": framework.NewStatus(framework.Unschedulable, "node(s) didn't fit the combination of plugin(s) [FakePreFilter2 FakePreFilter3]"),
+					},
+					UnschedulablePlugins: sets.String{},
+				},
+			},
+		},
+		{
+			name: "test prefilter plugin returning empty node set",
+			registerPlugins: []st.RegisterPluginFunc{
+				st.RegisterQueueSortPlugin(queuesort.Name, queuesort.New),
+				st.RegisterPreFilterPlugin(
+					"FakePreFilter1",
+					st.NewFakePreFilterPlugin("FakePreFilter1", nil, nil),
+				),
+				st.RegisterPreFilterPlugin(
+					"FakePreFilter2",
+					st.NewFakePreFilterPlugin("FakePreFilter2", &framework.PreFilterResult{NodeNames: sets.NewString()}, nil),
 				),
 				st.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
 			},
-			nodes:         []string{"1", "2"},
-			pod:           &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "test-prefilter", UID: types.UID("test-prefilter")}},
-			expectedHosts: nil,
-			wErr:          fmt.Errorf(`running PreFilter plugin "FakePreFilter": %w`, errors.New("injected error status")),
+			nodes: []string{"node1"},
+			pod:   &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "test-prefilter", UID: types.UID("test-prefilter")}},
+			wErr: &framework.FitError{
+				Pod:         &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "test-prefilter", UID: types.UID("test-prefilter")}},
+				NumAllNodes: 1,
+				Diagnosis: framework.Diagnosis{
+					NodeToStatusMap: framework.NodeToStatusMap{
+						"node1": framework.NewStatus(framework.Unschedulable, "node(s) didn't fit the combination of plugin(s) [FakePreFilter2]"),
```

Comment:
I think this looks confusing. It should say something like:

`node(s) didn't pass FakePrefilter2`

Only when a combination of 2 don't pass, we should say "combination" or "simultaneously".

Reply:
I doubt this will be triggered or common enough case to special case it, but I just did.